### PR TITLE
[Fix-9938][master] Fix 9938 query environment config set to task instance

### LINKED ISSUE #9938: [Bug] [server master] need fault tolerance task environment_config is null
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

need fault tolerance task  environment_config is null task execute failed
image

### What you expected to happen

image

### How to reproduce

1

### Anything else

_No response_

### Version

2.0.5

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by someorz**
check exists task environment code if not default query environment config set to task instance

## Brief change log
close https://github.com/apache/dolphinscheduler/issues/9938

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by SbloodyS** [comment]
Hi @someorz , please associate the issue.

**Comment by someorz** [comment]
@SbloodyS i had associate the issue.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java [hunk 1]
```diff
@@ -1115,6 +1115,12 @@ private TaskInstance findTaskIfExists(Long taskCode, int taskVersion) {
     private TaskInstance createTaskInstance(ProcessInstance processInstance, TaskNode taskNode) {
         TaskInstance taskInstance = findTaskIfExists(taskNode.getCode(), taskNode.getVersion());
         if (taskInstance != null) {
+            if (!taskInstance.getEnvironmentCode().equals(-1L)) {
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java:1118]:
```suggestion
            if (!taskInstance.getEnvironmentCode().equals(Constants.DEFAULT_ENVIRONMENT_CODE)) {
```

**Comment by ruanwenjun** [comment]
I find we already have this logic in `newTaskInstance`, so there is no need to do this in everywhere, am I right?

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java [hunk 2]
```diff
@@ -1115,6 +1115,12 @@ private TaskInstance findTaskIfExists(Long taskCode, int taskVersion) {
...
+                Environment environment = processService.findEnvironmentByCode(taskInstance.getEnvironmentCode());
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java:1119]: I think we should not query again after workflow running, and I found that there was already in `WorkflowExecuteRunnable.newTaskInstance`.

Comment (someorz) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java:1119]: This bug will not appear in new tasks. This bug is that existing tasks will appear during fault tolerance.
` TaskInstance taskInstance = findTaskIfExists(taskNode.getCode(), taskNode.getVersion());`
findTaskIfExists return instance EnvironmentConfig is null at version 2.0.5 if new version findTaskIfExists method return TaskInstance EnvironmentConfig is not null no need to query again

**Comment by someorz** [comment]
> I find we already have this logic in `newTaskInstance`, so there is no need to do this in everywhere, am I right?

This bug will not appear in new tasks. This bug is that existing tasks will appear during fault tolerance.
TaskInstance taskInstance = findTaskIfExists(taskNode.getCode(), taskNode.getVersion());
findTaskIfExists return instance EnvironmentConfig is null at version 2.0.5 if new version findTaskIfExists method return TaskInstance EnvironmentConfig is not null no need to query again

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java (hunk 2 above)
Comment (caishunfeng) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java:1119]: > This bug will not appear in new tasks. This bug is that existing tasks will appear during fault tolerance. ` TaskInstance taskInstance = findTaskIfExists(taskNode.getCode(), taskNode.getVersion());` findTaskIfExists return instance EnvironmentConfig is null at version 2.0.5 if new version findTaskIfExists method return TaskInstance EnvironmentConfig is not null no need to query again

In this case, we can add this logic into `initTaskQueue`, which will load the valid tasks at first if exist. What do you think?

Comment (someorz) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java:1119]: great is a more elegant solution，i think in baseSql add environment_config column will resolve the problem
image

**Pull request closed without merging**